script.js

```diff
@@ -242,4 +242,41 @@ function projectDetailWindowOpenClose(status) {
     document.querySelector('.project-details-bkg').classList.add('top-open');
   }
   projectDetailWindow(status);
-}
\ No newline at end of file
+}
+
+const button = document.querySelector('.form-button');
+button.addEventListener('click', (event) => {
+  const email = document.getElementById('email-address');
+  const emailAddress = email.value;
+  const lowerCase = emailAddress.toLowerCase();
+  if (emailAddress !== lowerCase) {
+    document.querySelector('.error').classList.add('display-flex');
+    event.preventDefault();
+  } else {
+    document.querySelector('.error').classList.remove('display-flex');
+  }
+});
+
+const userName = document.querySelector('#user-name');
+const emailAddress = document.querySelector('#email-address');
+const message = document.querySelector('#text-area');
+
+function dataStorage() {
+  const formData = {
+    name: userName.value,
+    email: emailAddress.value,
+    message: message.value,
+  };
+  localStorage.setItem('data', JSON.stringify(formData));
+}
+
+const elements = [userName, emailAddress, message];
+for (let i = 0; i < elements.length; i += 1) {
+  elements[i].addEventListener('focusout', dataStorage);
+}
+const userData = JSON.parse(localStorage.getItem('data'));
+if (userData) {
+  userName.value = userData.name;
+  emailAddress.value = userData.email;
+  message.value = userData.message;
+}
```

Comment:
@muhammaduzairrazaq I can see you adding a function for local storage but I cannot see where are you calling this function and I also cannot see the data in the browser. Please revisit this pull request and make sure the data will be saved in the browser: 
YOUR STORAGE:
![uzair](https://github.com/muhammaduzairrazaq/portfolio/assets/122353030/626b5d7d-adac-4c04-8801-946d80022cfc)

How
![hamza](https://github.com/muhammaduzairrazaq/portfolio/assets/122353030/e354edb4-e6e4-409b-ac3c-225ff2d4cb48)
 it should look like: